packages/utils/src/svg-css-sprite/create-sprite.ts

```diff
@@ -0,0 +1,212 @@
+import {
+	AsyncSpriteIcons,
+	AsyncSpriteIconsFactory,
+	SpriteCollection,
+	SpriteEntry,
+	SpriteIcon,
+	SpriteIcons,
+	Sprites,
+} from './types';
+import { yellow } from 'kolorist';
+import { loadNodeIcon } from '../loader/node-loader';
+import { ReadableStream } from 'node:stream/web';
+
+const warned = new Set<string>();
+
+export function createSprite(
+	spriteName: string,
+	icons: SpriteIcons,
+	warn = true
+) {
+	return `<svg xmlns="http://www.w3.org/2000/svg">${
+		Object.entries(typeof icons === 'function' ? icons() : icons).reduce(
+			(acc, [name, svg]) => {
+				const data = parseSVGData(spriteName, name, svg, warn);
+				if (data) {
+					acc.content += generateSpriteEntry(acc, data, {
+						name,
+						svg,
+					});
+				}
+
+				return acc;
+			},
+			<Sprites>{ content: '' }
+		).content
+	}
+</svg>`;
+}
+
+export function createAndPipeReadableStreamSprite(
+	spriteName: string,
+	icons: AsyncSpriteIcons | AsyncSpriteIconsFactory,
+	writableStream: WritableStream,
+	warn = true,
+	options?: StreamPipeOptions
+) {
+	return createReadableStreamSprite(spriteName, icons, warn).pipeTo(
+		writableStream,
+		options
+	);
+}
+
+export function createReadableStreamSprite(
+	spriteName: string,
+	icons: AsyncSpriteIcons | AsyncSpriteIconsFactory,
+	warn = true
+) {
+	const context: Sprites = { content: '' };
```

Comment:
Allow provide context, maybe we'll need to add icons and mappings.

Update other functions here and in nide module.

Reply:
Maybe we don't need this, the main problem will be on UnoCSS: when building we'll need to emit the svg sprite (will include a hash) and the css icon url style will need the sprite name.